llm-lora-finetuning/steps/evaluate_model.py

```diff
@@ -0,0 +1,85 @@
+# Apache Software License 2.0
+#
+# Copyright (c) ZenML GmbH 2024. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pathlib import Path
+from typing import Optional
+
+import evaluate
+import torch
+from datasets import load_from_disk
+from utils.loaders import (
+    load_base_model,
+    load_pretrained_model,
+    load_tokenizer,
+)
+from utils.tokenizer import tokenize_for_eval
+from zenml import log_model_metadata, save_artifact, step
+from zenml.logger import get_logger
+
+logger = get_logger(__name__)
+
+
+@step
+def evaluate_model(
```

Comment:
maybe a docstring for this and other steps explaining what they do?

Reply:
Added